core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -303,6 +287,325 @@ public static Properties filterAndRemovePrefix(Map<String, String> properties, S
+    return String.format(LIST_TABLES_OR_VIEWS_SQL, tableOrViewTableName, tableOrViewNamespace);
+  }
+
+  public static String listTablesSql() {
+    return listTablesOrViewsSql(true);
+  }
+
+  public static String listViewsSql() {
+    return listTablesOrViewsSql(false);
+  }
+
+  private static String renameTableOrViewSql(boolean table) {
+    String tableOrViewTableName = table ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String tableOrViewNamespace = table ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String tableOrViewName = table ? TABLE_NAME : VIEW_NAME;
+    return String.format(
+        RENAME_TABLE_OR_VIEW_SQL,
+        tableOrViewTableName,
+        tableOrViewNamespace,
+        tableOrViewName,
+        tableOrViewNamespace,
+        tableOrViewName);
+  }
+
+  public static String renameTableSql() {
+    return renameTableOrViewSql(true);
+  }
+
+  public static String renameViewSql() {
+    return renameTableOrViewSql(false);
+  }
+
+  private static String dropTableOrViewSql(boolean table) {
+    String tableOrViewTableName = table ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String tableOrViewNamespace = table ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String tableOrViewName = table ? TABLE_NAME : VIEW_NAME;
+    return String.format(
+        DROP_TABLE_OR_VIEW_SQL, tableOrViewTableName, tableOrViewNamespace, tableOrViewName);
+  }
+
+  public static String dropTableSql() {
+    return dropTableOrViewSql(true);
+  }
+
+  public static String dropViewSql() {
+    return dropTableOrViewSql(false);
+  }
+
+  private static String doCommitCreateTableOrViewSql(boolean table) {
+    String tableOrViewTableName = table ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String tableOrViewNamespace = table ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String tableOrViewName = table ? TABLE_NAME : VIEW_NAME;
+    return String.format(
+        DO_COMMIT_CREATE_TABLE_OR_VIEW_SQL,
+        tableOrViewTableName,
+        tableOrViewNamespace,
+        tableOrViewName);
+  }
+
+  private static int updateTableOrView(
+      boolean table,
+      JdbcClientPool connections,
+      String catalogName,
+      TableIdentifier tableOrViewIdentifier,
+      String newMetadataLocation,
+      String oldMetadataLocation)
+      throws SQLException, InterruptedException {
+    return connections.run(
+        conn -> {
+          try (PreparedStatement sql = conn.prepareStatement(doCommitSql(table))) {
+            // UPDATE
+            sql.setString(1, newMetadataLocation);
+            sql.setString(2, oldMetadataLocation);
+            // WHERE
+            sql.setString(3, catalogName);
+            sql.setString(4, namespaceToString(tableOrViewIdentifier.namespace()));
+            sql.setString(5, tableOrViewIdentifier.name());
+            sql.setString(6, oldMetadataLocation);
+            return sql.executeUpdate();
+          }
+        });
+  }
+
+  public static int updateTable(
```

Comment:
we should make as much of these methods as possible package-private rather than public